### PR TITLE
Build Linux packages for the extension

### PR DESCRIPTION
**Pull request by SergeyKleyman**
Closes open-telemetry/opentelemetry-php#1151

### CODE REVIEW

File: packaging/before-uninstall.sh [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+function php_command() {
+    PHP_BIN=$(command -v php)
+    ${PHP_BIN} -d memory_limit=128M "$@"
+}
+
+################################################################################
+#### Function php_ini_file_path ################################################
+function php_ini_file_path() {
+    php_command -i \
+        | grep 'Configuration File (php.ini) Path =>' \
```
Comment (brettmc) [inline packaging/before-uninstall.sh:29]: could this be replaced by `php-config --ini-path` ?

File: packaging/before-uninstall.sh [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+function php_api() {
+    php_command -i \
+        | grep 'PHP API' \
+        | sed -e 's#.* =>##g' \
+        | awk '{print $1}'
+}
+
+################################################################################
+#### Function php_config_d_path ################################################
+function php_config_d_path() {
```
Comment (brettmc) [inline packaging/before-uninstall.sh:46]: `php-config --ini-dir` ?

File: packaging/before-uninstall.sh [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+}
+
+################################################################################
+#### Function uninstall_conf_d_files ###########=###############################
+function uninstall_conf_d_files() {
+    PHP_CONFIG_D_PATH=$1
+
+    echo "Uninstalling ${OPENTELEMETRY_INI_FILE_NAME} for supported SAPI's"
+
+    # Detect installed SAPI's
```
Comment (brettmc) [inline packaging/before-uninstall.sh:85]: `php-config --php-sapis` ?

File: packaging/create-package.sh [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+BUILD_EXT_DIR=""
+
+if [ "${TYPE}" = 'apk' ] ; then
+	BUILD_EXT_DIR=instrumentation/native/_build/linuxmusl-x86-64-release/ext/
+else
+	BUILD_EXT_DIR=instrumentation/native/_build/linux-x86-64-release/ext/
+fi
+
+touch build/open-telemtry.ini
```
Comment (brettmc) [inline packaging/create-package.sh:17]:
```suggestion
touch build/open-telemetry.ini
```

File: packaging/create-package.sh [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+
+# create second tar for musl
+if [ "${TYPE}" = 'tar' ] ; then
+	NAME_BACKUP=${NAME}
+	NAME="${NAME_BACKUP}-linux-x86-64"
+	BUILD_EXT_DIR=instrumentation/native/_build/linux-x86-64-release/ext/
+	createPackage
+
+	NAME="${NAME_BACKUP}-debugsymbols-linux-x86-64"
```
Comment (brettmc) [inline packaging/create-package.sh:97]: minor, but I think just `debug` would be ok here?

File: packaging/post-install.sh [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+
+################################################################################
+############################ GLOBAL VARIABLES ##################################
+################################################################################
+PHP_INSTRUMENTATION_DIR=/opt/open-telemtry/php-instrumentation
+EXTENSION_DIR="${PHP_INSTRUMENTATION_DIR}/extensions"
+EXTENSION_CFG_DIR="${PHP_INSTRUMENTATION_DIR}/etc"
+BOOTSTRAP_FILE_PATH="${PHP_INSTRUMENTATION_DIR}/src/bootstrap_php_part.php"
+BACKUP_EXTENSION=".instrumentation.bck"
+OPENTELEMETRY_INI_FILE_NAME="open-telemtry.ini"
```
Comment (brettmc) [inline packaging/post-install.sh:14]:
```suggestion
OPENTELEMETRY_INI_FILE_NAME="open-telemetry.ini"
```

File: packaging/post-install.sh [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+######### Let's support alpine installations
+PATH=${PATH}:/usr/local/bin
+
+################################################################################
+############################ GLOBAL VARIABLES ##################################
+################################################################################
+PHP_INSTRUMENTATION_DIR=/opt/open-telemtry/php-instrumentation
+EXTENSION_DIR="${PHP_INSTRUMENTATION_DIR}/extensions"
+EXTENSION_CFG_DIR="${PHP_INSTRUMENTATION_DIR}/etc"
+BOOTSTRAP_FILE_PATH="${PHP_INSTRUMENTATION_DIR}/src/bootstrap_php_part.php"
```
Comment (brettmc) [inline packaging/post-install.sh:12]: I think this is vestigial from elastic apm, and not needed here

File: packaging/before-uninstall.sh [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+        | grep 'Configuration File (php.ini) Path =>' \
+        | sed -e 's#Configuration File (php.ini) Path =>##g' \
+        | head -n 1 \
+        | awk '{print $1}'
+}
+
+################################################################################
+#### Function php_api ##########################################################
+function php_api() {
+    php_command -i \
```
Comment (brettmc) [inline packaging/before-uninstall.sh:38]: if there was a `php-config --api`, we could get everything more effectively from there...

**Review by brettmc — COMMENTED** [review]
This looks great! A few comments and observations from me. Do you suggest that we should run this build as part of each push, and also do the publish when we create a release? If so, we will also need some actions added under .github

**Comment by brettmc** [comment]
Hopefully this will integrate nicely with #109 when it's merged - we can publish these packages as build artifacts for each github actions run, and also as release artifacts.

**Comment by bobstrecansky** [comment]
Going to close this for now; please reopen if you're interested in finishing this out.

**Pull request closed without merging**